test/e2e/kubernetes/kubernetes_test.go

```diff
@@ -704,6 +704,13 @@ var _ = Describe("Azure Container Cluster using the Kubernetes Orchestrator", fu
 			By("Ensuring that we have functional DNS resolution from a linux container")
 			validateDNSLinuxName := "validate-dns-linux"
 			validateDNSLinuxNamespace := "default"
+			// Create a kubectl version check to avoid api version mismatch which due to deployment failure in next step
```

Comment:
FYI, this is fine as an informational check, but the `k` util is already designed to use the version of `kubectl` that the API returns in a call to the cluster. Is that not working in your experience?

Reply:
In my local test with out ssh enabled, I found my kubectl version comes from https://github.com/Azure/aks-engine/blob/master/scripts/k#L40
It's usually older than test cluster's version which result in k create * failure. 
I didn't experience auto cluster call and match in local so I add this check. I have to manually override it. However, from pipeline validation, it seems kubectl version can match.